--- src/com/oltpbenchmark/benchmarks/jpab/JPABBenchmark.java
+++ src/com/oltpbenchmark/benchmarks/jpab/JPABBenchmark.java
@@ -49,5 +49,5 @@
         @Var Test test = null;
         try {
-            test = (Test)Class.forName("com.oltpbenchmark.benchmarks.jpab.tests."+this.jpabConf.getTestName()).newInstance();
+            test = (Test)Class.forName("com.oltpbenchmark.benchmarks.jpab.tests."+this.jpabConf.getTestName()).getDeclaredConstructor().newInstance();
             int totalObjectCount= (int) this.workConf.getScaleFactor();
             test.setBatchSize(1);
